utils/utils.go

```diff
@@ -140,3 +142,16 @@ func ProgressBar(prefix string, size int64, onComplete string) (*mpb.Progress, *
 
 	return p, bar
 }
+
+// CheckMaxMemory gets the total system memory and compares it to the variable.  if the variable
+// is larger than the total memory, it returns an error
+func CheckMaxMemory(newMem strongunits.MiB) error {
+	memStat, err := mem.VirtualMemory()
+	if err != nil {
+		return err
+	}
+	if total := strongunits.B(memStat.Total); strongunits.B(memStat.Total) < newMem.ToBytes() {
+		return fmt.Errorf("requested amount of memory (%d MB) greater than total memory (%d MB)", newMem, total)
```

Comment:
Maybe say `total system memory`? Otherwise it may not be clear what total we are referring to.